crates/primitives/transactions/src/getters.rs

```diff
@@ -1,331 +1,206 @@
-            DeclareTransaction::V0(_) => None,
-            DeclareTransaction::V1(_) => None,
-            DeclareTransaction::V2(tx) => Some(&tx.compiled_class_hash),
-        }
-    }
-
-    pub fn offset_version(&self) -> bool {
-        match self {
-            // we don't accept V0 txs from the RPC
-            DeclareTransaction::V0(_) => false,
-            DeclareTransaction::V1(tx) => tx.offset_version,
-            DeclareTransaction::V2(tx) => tx.offset_version,
-        }
-    }
-}
-
-impl DeployAccountTransaction {
-    pub fn signature(&self) -> &Vec<Felt252Wrapper> {
-        &self.signature
-    }
-
-    pub fn max_fee(&self) -> &u128 {
-        &self.max_fee
-    }
-
-    pub fn calldata(&self) -> &Vec<Felt252Wrapper> {
-        &self.constructor_calldata
-    }
-
-    pub fn nonce(&self) -> &Felt252Wrapper {
-        &self.nonce
-    }
+use crate::TxType;
 
-    pub fn account_address(&self) -> Felt252Wrapper {
-        Felt252Wrapper(self.get_account_address())
-    }
-
-    pub fn class_hash(&self) -> &Felt252Wrapper {
-        &self.class_hash
-    }
-
-    pub fn offset_version(&self) -> bool {
-        self.offset_version
-    }
+pub trait Getters {
+    fn sender_address(&self) -> Felt252Wrapper;
+    fn signature(&self) -> Vec<Felt252Wrapper>;
+    fn calldata(&self) -> Option<Vec<Felt252Wrapper>>;
+    fn nonce(&self) -> Option<Felt252Wrapper>;
+    fn tx_type(&self) -> TxType;
 }
 
-impl DeployTransaction {
-    pub fn calldata(&self) -> &Vec<Felt252Wrapper> {
-        &self.constructor_calldata
-    }
-
-    pub fn version(&self) -> u8 {
-        1
-    }
-
-    pub fn account_address(&self) -> Felt252Wrapper {
-        Felt252Wrapper(self.get_account_address())
-    }
-
-    pub fn class_hash(&self) -> &Felt252Wrapper {
-        &self.class_hash
-    }
+pub trait Hash {
+    fn tx_hash(&self) -> Option<TransactionHash>;
 }
 
-impl InvokeTransaction {
-    pub fn sender_address(&self) -> &Felt252Wrapper {
-        match self {
-            InvokeTransaction::V0(tx) => &tx.contract_address,
-            InvokeTransaction::V1(tx) => &tx.sender_address,
-        }
-    }
-
-    pub fn signature(&self) -> &Vec<Felt252Wrapper> {
-        match self {
-            InvokeTransaction::V0(tx) => &tx.signature,
-            InvokeTransaction::V1(tx) => &tx.signature,
-        }
-    }
-
-    pub fn max_fee(&self) -> &u128 {
-        match self {
-            InvokeTransaction::V0(tx) => &tx.max_fee,
-            InvokeTransaction::V1(tx) => &tx.max_fee,
-        }
-    }
-
-    pub fn calldata(&self) -> &Vec<Felt252Wrapper> {
+impl Getters for AccountTransaction {
+    fn sender_address(&self) -> Felt252Wrapper {
         match self {
-            InvokeTransaction::V0(tx) => &tx.calldata,
-            InvokeTransaction::V1(tx) => &tx.calldata,
+            AccountTransaction::Declare(tx) => tx.tx.sender_address().into(),
+            AccountTransaction::DeployAccount(tx) => tx.tx.contract_address_salt().into(),
```

Comment:
Is it correct that `contract_address_salt` is the sender address or was this a misstype?